File: 프로그래머스/1/134240. 푸드 파이트 대회/푸드 파이트 대회.js
function solution(food) {
    var answer = '';
    let halfarr = [];
    
    for (i = 1; i<food.length; i++) {
        for(k=1; k <= Math.floor(food[i]/2); k++){
            halfarr.push(i);
        }
    }
    
    answer = halfarr.join("").concat("0",halfarr.reverse().join(""));
    return answer;
}

File: 프로그래머스/1/134240. 푸드 파이트 대회/푸드 파이트 대회.java
import java.util.*;

class Solution {
    public String solution(int[] food) {
        String answer = "";
        
        StringBuilder sb = new StringBuilder();
        
        for(int i=1; i<food.length; i++){
            for(int j=1; j<=food[i]/2; j++){
              sb.append(i);  
            }
        }
        System.out.println(sb);

        String left = sb.toString();
        String right = sb.reverse().toString();
        answer = left+ "0" + right;
        
        return answer;
    }
}